clang/lib/Driver/Driver.cpp

```diff
@@ -4791,6 +4791,9 @@ void Driver::BuildActions(Compilation &C, DerivedArgList &Args,
     if (!UnbundlerInputs.empty()) {
       Action *PartialLink =
           C.MakeAction<PartialLinkJobAction>(UnbundlerInputs, types::TY_Object);
+      if (!LastArg)
+        LastArg =
+            &(dyn_cast<InputAction>(UnbundlerInputs.back())->getInputArg());
```

Comment:
Are we guaranteed that the arg being used from UnbundlerInputs.back() is an InputAction?

Reply:
Yes it does, Just verified again. The test pass the input arg to linker properly.